core/google/cloud/exceptions.py

```diff
@@ -20,16 +20,26 @@
 # Avoid the grpc and google.cloud.grpc collision.
 from __future__ import absolute_import
 
+import contextlib
 import copy
+import sys
 
 import six
 
-from google.cloud._helpers import _to_bytes
-
 try:
+    from google.gax.errors import GaxError
+    from google.gax.grpc import exc_to_code
+    from grpc import StatusCode
     from grpc._channel import _Rendezvous
 except ImportError:  # pragma: NO COVER
+    _HAVE_GRPC = False
     _Rendezvous = None
+else:
+    _HAVE_GRPC = True
```

Comment:
De gustibus?  ISTM that the `else:` block of a `try:` is where everything goes which is *not* going to trigger one of the handled exceptions.